model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -177,17 +177,17 @@ object AddressResponseAddress {
     val trimmedDependentThoroughfare = paf.dependentThoroughfare.trim
     val trimmedThoroughfare = paf.thoroughfare.trim
 
-    val buildingNumberWithStreetName = if (trimmedDependentThoroughfare.isEmpty) {
-      s"$trimmedBuildingNumber $trimmedThoroughfare"
-    } else {
-      s"$trimmedBuildingNumber $trimmedDependentThoroughfare, $trimmedThoroughfare"
-    }
+    val buildingNumberWithStreetName =
+      s"$trimmedBuildingNumber ${ if(trimmedDependentThoroughfare.nonEmpty) s"$trimmedDependentThoroughfare, " else "" }$trimmedThoroughfare"
 
-    Seq(paf.departmentName, paf.organizationName, paf.subBuildingName, paf.buildingName,
+    val parts = Seq(paf.departmentName, paf.organizationName, paf.subBuildingName, paf.buildingName,
       poBoxNumber, buildingNumberWithStreetName, paf.doubleDependentLocality, paf.dependentLocality,
-      paf.postTown, paf.postcode).map(_.trim).filter(_.nonEmpty).mkString(", ")
+      paf.postTown, paf.postcode)
+
+    delimitByComma(parts)
   }
 
+  private def delimitByComma(parts: Seq[String]) = parts.map(_.trim).filter(_.nonEmpty).mkString(", ")
```

Comment:
Nice one, no need for `Seq` here, you can use varargs 

eg: 
`parts: String *`

It saves you making Seq's, and you can directly invoke